### PR TITLE
fix bugs and auto-import

### PR DESCRIPTION
**Pull request by skeeey**
1. using registration auto-import
2. waiting for the agent is registered
3. fix agent addon permission errors in hosted mode
4. ignore work manager log cert 
5. build in managed-serviceaccount addon
6. refactor e2e

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -6,20 +6,18 @@ import (
...
+
+	"open-cluster-management.io/multicluster-controlplane/pkg/features"
 
 	"github.com/stolostron/multicluster-controlplane/cmd/agent"
 	"github.com/stolostron/multicluster-controlplane/cmd/manager"
 	"github.com/stolostron/multicluster-controlplane/pkg/feature"
 )
 
 func init() {
-	utilruntime.Must(logsapi.AddFeatureGates(utilfeature.DefaultMutableFeatureGate)) // register log to featuregate
```
Comment (clyang82) [inline cmd/main.go:20]: Do we need register log?

File: hack/deploy/agent/deployment.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ spec:
           - "agent"
           - "--cluster-name=cluster1"
           - "--bootstrap-kubeconfig=/spoke/bootstrap/kubeconfig"
-          - "--feature-gates=AddonManagement=true"
+          - "--feature-gates=ManagedServiceAccount=false"
```
Comment (clyang82) [inline hack/deploy/agent/deployment.yaml:28]: the default value is false. we may remove `--feature-gates` for now.

File: pkg/agent/addons/managedserviceaccount.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
+package addons
+
+import (
+	"context"
+	"fmt"
+
+	khelpers "github.com/stolostron/multicluster-controlplane/pkg/controllers/klusterlet/helpers"
```
Comment (clyang82) [inline pkg/agent/addons/managedserviceaccount.go:7]: the import order should be:
```
golang packages

third-party packages

the current packages
```

File: pkg/agent/agent.go [hunk 4]
```diff
@@ -191,6 +196,13 @@ func (a *AgentOptions) RunAddOns(ctx context.Context) error {
 		startCtrlMgr = true
 	}
 
+	if features.DefaultAgentMutableFeatureGate.Enabled(feature.ManagedServiceAccount) {
+		klog.Info("starting managed serviceaccount addon agent")
+		if err := addons.StartManagedServiceAccountAgent(ctx, hubManager, clusterName); err != nil {
+			klog.Fatalf("failed to setup managed serviceaccount addon, %v", err)
+		}
+	}
```
Comment (clyang82) [inline pkg/agent/agent.go:204]: need `startCtrlMgr = true` here

File: pkg/controllers/addons/managedclusterinfo.go [hunk 5]
```diff
@@ -8,11 +8,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
-// TODO update work manager to support this as option
-const logCertSecret = "open-cluster-management/ocm-klusterlet-self-signed-secrets"
-
 func SetupManagedClusterInfoWithManager(ctx context.Context, mgr manager.Manager) error {
-	if err := clusterinfo.SetupWithManager(mgr, logCertSecret); err != nil {
+	if err := clusterinfo.SetupWithManager(mgr, ""); err != nil {
```
Comment (clyang82) [inline pkg/controllers/addons/managedclusterinfo.go:12]: 👍

File: pkg/controllers/clusterimport/manager.go [hunk 6]
```diff
@@ -15,39 +14,11 @@ import (
...
 	"sigs.k8s.io/controller-runtime/pkg/event"
-	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
-	runtimesource "sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-const controllerName = "autoimport-controller"
-
-func SetupCSRApprovalController(mgr manager.Manager, kubeClient kubernetes.Interface) error {
```
Comment (clyang82) [inline pkg/controllers/clusterimport/manager.go:26]: no CSR required?

File: pkg/controllers/klusterlet/manifests/klusterlet/management/klusterlet-agent-deployment.yaml [hunk 7]
```diff
@@ -26,10 +26,15 @@ spec:
           - "--cluster-name={{ .ClusterName }}"
```
Comment (clyang82) [inline pkg/controllers/klusterlet/manifests/klusterlet/management/klusterlet-agent-deployment.yaml:26]: Do we start this deployment?

File: cmd/main.go (hunk 1 above)
Comment (skeeey) [reply cmd/main.go:20]: it seems we need not, if we want it, we can enable it on the ocm part

File: pkg/agent/addons/managedserviceaccount.go (hunk 3 above)
Comment (skeeey) [reply pkg/agent/addons/managedserviceaccount.go:7]: reordered

File: pkg/agent/agent.go (hunk 4 above)
Comment (skeeey) [reply pkg/agent/agent.go:204]: added

File: pkg/controllers/clusterimport/manager.go (hunk 6 above)
Comment (skeeey) [reply pkg/controllers/clusterimport/manager.go:26]: we don't need need this any more, the ocm controlplane agent will handle this

File: pkg/controllers/klusterlet/manifests/klusterlet/management/klusterlet-agent-deployment.yaml (hunk 7 above)
Comment (skeeey) [reply pkg/controllers/klusterlet/manifests/klusterlet/management/klusterlet-agent-deployment.yaml:26]: the klusterlet controller will apply this to the management cluster to deploy the agent with hosted mode

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**